packages/app/src/components/Settings/Settings.js

```diff
@@ -0,0 +1,267 @@
+            
+            setGcToEdit({openedOn: "", closedOn: "", name: ""})
+            getGrantCycles();
+            setShowEditModal(false);
+        }
+        catch(ex) {
+            if (ex.response && ex.response.status === 400) {
+                setEditErrors(ex.response.data);
+            }
+        }
+    }
+
+    const handleEdit = gc => {
+        setGcToEdit({ openedOn: gc.openedOn.split("T")[0], closedOn: gc.closedOn.split("T")[0], name: gc.name, id: gc.id });
+        setShowEditModal(true);
+    }
+
+    useEffect(() => {
+        handleDisableButton(newGrantCycle);
+    }, [newGrantCycle])
+
+    useEffect(() => {
+        if (gcToEdit)
+            handleDisableEditButton(gcToEdit);
+    }, [gcToEdit])
+
+    const handleDisableButton = (gc) => {
+        if (gc.openedOn !== "" && gc.closedOn !== "") {
+            const millisecondsInADay = 86400000;
+            const startDate = new Date(gc.openedOn);
+            const endDate = new Date(gc.closedOn);
+            const result = endDate - startDate >= millisecondsInADay ? false : true;
+            
+            setDisableButton(result);
+            if (!result) {
+                setErrors("")
+                if (gc.name === "") {
+                    setDisableButton(true);
+                    setErrors("Please enter a name for the grant cycle");
+                }
+            }
+            else setErrors("Start Date must be earlier than End Date");
+        }
+        else {
+            setDisableButton(true);
+        }
+    }
+    
+    const handleDisableEditButton = (gc) => {
+        if (gc.openedOn !== "" && gc.closedOn !== "") {
+            const millisecondsInADay = 86400000;
+            const startDate = new Date(gc.openedOn);
+            const endDate = new Date(gc.closedOn);
+            const result = endDate - startDate >= millisecondsInADay ? false : true;
+            
+            setDisableEditButton(result);
+            if (!result) {
+                setEditErrors("")
+                if (gc.name === "") {
+                    setDisableEditButton(true);
+                    setEditErrors("Please enter a name for the grant cycle");
+
+                }
+                const closed = moment(gc.closedOn);
+                if (moment().isAfter(closed))
+                    setDisableEditButton(true);
+            }
+            else setEditErrors("Start Date must be earlier than End Date");
+        }
+        
+        else {
+            setDisableEditButton(true);
+        }
+
+        if (disableDatePicker(gc.closedOn)) {
+            setEditErrors("Cannot set date earlier than today.");
+            setDisableEditButton(true);
+        }
+    }
+
+    const closeEditModal = () => {
+        setShowEditModal(false);
+    }
+
+    const disableDatePicker = (d) => {
+        return moment().isAfter(moment(d));
+    }
+
+    const formatDateString = date => {
+        const year = date.slice(0, 4);
```

Comment:
This function is a duplicate of another one above. This should be refactored into a helper method.